### PR TITLE
Bringing Evolution to the Job Engine Task library developed in the workload project to master

### PR DESCRIPTION
**Pull request by samcake**
part of work item [ 14786](https://highfidelity.fogbugz.com/f/cases/14786/Workload-Project-Physics-graceful-failure)
This PR brings some of the improvements and refactors we did to the Job Engine Task system in the task lib during the Workload project into master.
This work is not suppose to change the behavior but clean up some of the default activations of some jobs that should be off (debug jobs), expose in the Config /jsapi of the Job more features to navigate the Task graph and a reusable UI/qml component and lib to inspect an engine.

In particular:
- All the time measurement is now relying on the c++ chrono library
- Define the Engine class in the task lib which is then derived from by the render::RenderEngine class
- Rename the ex render::Engine class to render::RenderEngine
- Templatize the time profiling class used to  instrument the tasks
- On Config expose the js api calls to be able to navigate the graph of tasks
- turned off 2 debug Jobs that were on

## TEST PLAN
Smoke test everything should work the same at the same performances

Run the Luci.js from the installed scripts folder (in developer/utilities/render)
The tool should work as usual
and present a new button at the bottom called "engine"
pressing that button shows a tree list view of the graph of jobs in the engine
you can fold/unfold task and find the sub tasks running in it
The actual timing of each Job is displayed in real time.

### CODE REVIEW

File: scripts/developer/utilities/lib/jet/jet.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        printout(depthTab + index + " " + job.objectName + " " + (job.enabled ? "on " : "off ") + job.cpuRunTime + "ms") 
+        if (showProps) {
+            var keys = job_propKeys(job);
+            for (var p=0; p < keys.length;p++) {
+                var prop = job[keys[p]]
+                printout(depthTab + tab + tab + typeof prop + " " + keys[p] + " " + prop);
+            }
+        }
+        return true
+     //   return depth < maxDepth;
```
Comment (AndrewMeadows) [inline scripts/developer/utilities/lib/jet/jet.js]: Commented out cruft?

File: scripts/developer/utilities/lib/jet/qml/TaskListView.qml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+import "../jet.js" as Jet
+
+Rectangle {
+    HifiConstants { id: hifi;}
+    color: hifi.colors.baseGray;
+    id: root;
+
+ //   width: parent ? parent.width : 200
+ //   height: parent ? parent.height : 400
```
Comment (AndrewMeadows) [inline scripts/developer/utilities/lib/jet/qml/TaskListView.qml]: Commented out cruft?

File: libraries/task/src/task/Task.h [hunk 3]
```diff
@@ -370,15 +369,44 @@ class Task : public Job<JC> {
...
-#define Task_DeclareTypeAliases(ContextType) \
+#define Task_DeclareTypeAliases(ContextType, TimeProfiler) \
     using JobConfig = task::JobConfig; \
     using TaskConfig = task::TaskConfig; \
     template <class T> using PersistentConfig = task::PersistentConfig<T>; \
-    using Job = task::Job<ContextType>; \
-    using Task = task::Task<ContextType>; \
+    using Job = task::Job<ContextType, TimeProfiler>; \
+    using Task = task::Task<ContextType, TimeProfiler>; \
+    using _Engine = task::Engine<ContextType, TimeProfiler>; \
```
Comment (AndrewMeadows) [inline libraries/task/src/task/Task.h]: Yeah, this `_Engine` name choice is too weird.  Make it `EngineBase` or something.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10050/HighFidelity-Beta-PR13219-55945b5454b5599801b43597eaf1728ab41b8358.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10050/HighFidelity-Beta-Interface-PR13219-55945b5454b5599801b43597eaf1728ab41b8358.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10064/HighFidelity-Beta-PR13219-e4048c8f2353b7cdd53393ba5a553b383c3dbac1.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10064/HighFidelity-Beta-Interface-PR13219-e4048c8f2353b7cdd53393ba5a553b383c3dbac1.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10068/HighFidelity-Beta-PR13219-7ed39f4052babf88b0dc91117940ebef6deb1e82.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10068/HighFidelity-Beta-Interface-PR13219-7ed39f4052babf88b0dc91117940ebef6deb1e82.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10072/HighFidelity-Beta-PR13219-c66b592b3a16ee512528ae1a68d417e9e49c82c5.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10072/HighFidelity-Beta-PR13219-c66b592b3a16ee512528ae1a68d417e9e49c82c5.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10072/HighFidelity-Beta-Interface-PR13219-c66b592b3a16ee512528ae1a68d417e9e49c82c5.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10072/HighFidelity-Beta-PR13219-c66b592b3a16ee512528ae1a68d417e9e49c82c5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10072/HighFidelity-Beta-Interface-PR13219-c66b592b3a16ee512528ae1a68d417e9e49c82c5.exe

**Comment by qligmaloney** [comment]
Currently testing

**Comment by qligmaloney** [comment]
PASS @conklin94122  No new defects. Script runs and behaves as expected

**Pull request merged**